### PR TITLE
NIFI-3373: Add nifi.flow.configuration.archive.max.count

### PR DESCRIPTION
**Pull request by ijokarumawak**
- Add 'nifi.flow.configuration.archive.max.count' in nifi.properties
- Change default archive limit so that it uses archive max time(30 days)
  and storage (500MB) if no limitation is specified
- Simplified logic to delete old archives

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java [hunk 1]
```diff
@@ -972,11 +973,15 @@ public Integer getFlowElectionMaxCandidates() {
     }
 
     public String getFlowConfigurationArchiveMaxTime() {
-        return getProperty(FLOW_CONFIGURATION_ARCHIVE_MAX_TIME, DEFAULT_FLOW_CONFIGURATION_ARCHIVE_MAX_TIME);
+        return getProperty(FLOW_CONFIGURATION_ARCHIVE_MAX_TIME, null);
```
Comment (joewitt) [inline nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java]: @ijokarumawak i think for each of these max values we should have some sensible default.  For time perhaps 14 days.  For storage perhaps 1GB.  For count 50.

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 2]
```diff
@@ -1934,9 +1934,10 @@ The first section of the _nifi.properties_ file is for the Core Properties. Thes
...
+|nifi.flow.configuration.archive.dir*|The location of the archive directory where backup copies of the flow.xml are saved. The default value is ./conf/archive. NiFi removes old archive files to limit disk usage based on file lifespan total size, and number of files, as specified with max.time, max.storage and max.count properties described below. This cleanup mechanism takes into account only automatically created archived flow.xml files. If there are other files or directories in this archive directory, NiFi will ignore them. Automatically created archives have filename with ISO 8601 format timestamp prefix followed by '_<original-filename>'. That is <year><month><day>T<hour><minute><second>+<timezone offset>_<original filename>. For example, `20160706T160719+0900_flow.xml.gz`. NiFi checks filenames when it cleans archive directory. If you would like to keep a particular archive in this directory without worrying about NiFi deleting it, you can do so by copying it with a different filename pattern.
```
Comment (joewitt) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc]: if you do agree we should provide sensible defaults we should add/indicate them in here.

Comment (ijokarumawak) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc]: Updated admin doc to describe how limitation settings work if no configuration is set for those.

**Comment by joewitt** [comment]
@ijokarumawak thanks for explaining and yes I'm totally on track now.  thanks.  Can you please squash and merge.  +1

**Comment by ijokarumawak** [comment]
Following Joe's +1, squashed and merged it into master.

**Pull request closed without merging**